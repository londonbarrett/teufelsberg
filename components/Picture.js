import * as React from 'react';
import PropTypes from 'prop-types';
import config from '../config';

const Picture = ({ uri, alt }) => (
  <div>
    <img
      className="image"
      src={`${config.imagePrefix}${uri}${config.bigSuffix}`}
      alt={alt}
    />
    <style jsx>
      {`
        .container {
          width: 100%;
        }
        .image {
          width: 100%;
        }
      `}
    </style>
  </div>
);

Picture.propTypes = {
  uri: PropTypes.string.isRequired,
  alt: PropTypes.string,
};

Picture.defaultProps = {
  alt: 'Mobile.de',
};

export default Picture;
