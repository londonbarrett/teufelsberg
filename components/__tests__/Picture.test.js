/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import Picture from '../Picture';

describe('Picture Tests', () => {
  it('Renders correctly', () => {
    const component = renderer.create(<Picture uri="picture_01" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
