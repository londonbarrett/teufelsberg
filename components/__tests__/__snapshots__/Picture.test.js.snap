// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture Tests Renders correctly 1`] = `
<div
  className="jsx-4014059616"
>
  <img
    alt="Mobile.de"
    className="jsx-4014059616 image"
    src="https://picture_01_27.jpg"
  />
</div>
`;
